import-automation/executor/app/service/github_api.py

```diff
@@ -0,0 +1,374 @@
+
+        Example:
+            Assume commit '12f23f1eeg234' to rhe repository modifies
+            'scripts/README.md', adds 'scripts/us_fed/README.md', and removes
+            'utils/foo.py'.
+
+            The method call
+                _query_changed_files_in_commit('12f23f1eeg234')
+            returns [
+                ('scripts/README.md', 'modified'),
+                ('scripts/us_fed/README.md', 'added'),
+                ('utils/foo.py', 'removed')
+            ]
+
+        Args:
+            commit_sha: Commit ID as a string.
+
+        Returns:
+            A list of tuples each of the form (<path>, <status>), where <path>
+            is the path to the file in the repository and <status> indicates
+            how the file is changed.
+        """
+        commit_info = self.query_commit(commit_sha)
+        files = []
+        for entry in commit_info['files']:
+            files.append((entry['filename'], entry['status']))
+        return files
+
+    def _query_files_in_dir(self, commit_sha: str, dir_path: str) -> List[str]:
+        """Queries the files in a directory at the state of a commit.
+
+        Only files are returned, not directories.
+
+        Args:
+            commit_sha: ID of the commit as a string.
+            dir_path: Path to the directory as a string, relative to the root
+                directory of the repository.
+
+        Returns:
+            List of relative paths to the files, each as a string.
+        """
+        content_query = _GITHUB_CONTENT_API.format_map({
+            'owner_username': self.owner,
+            'repo_name': self.repo,
+            'commit_sha': commit_sha,
+            'dir_path': dir_path
+        })
+        response = requests.get(content_query, auth=self.auth)
+        response.raise_for_status()
+        content_info = response.json()
+        return [
+            entry['name'] for entry in content_info if entry['type'] == 'file'
+        ]
+
+    def _dir_exists(self, commit_sha: str, dir_path: str) -> bool:
+        """Checks whether a directory still exists in the repository
+        after a commit.
+
+        Args:
+            commit_sha: Commit ID as a string
+            dir_path: Path to the directory to check as a string.
+
+        Returns:
+            True if the directory exists. False, otherwise.
+        """
+        content_query = self._build_content_query(commit_sha, dir_path)
+        response = requests.get(content_query, auth=self.auth)
+        is_ok = response.status_code == http.HTTPStatus.OK
+        is_not_found = response.status_code == http.HTTPStatus.NOT_FOUND
+        if not is_ok and not is_not_found:
+            response.raise_for_status()
+        return response.status_code != http.HTTPStatus.NOT_FOUND
+
+    def _path_containing_file(self, commit_sha: str, dir_path: str,
+                              containing: str) -> bool:
+        """Checks if the directory in the repository contains a file.
```

Comment:
I think this is direct containment, not containment within a subdirectory.

Reply:
Could you explain a bit more?